libs/experimental/langchain_experimental/prompt_injection_identifier/hugging_face_identifier.py

```diff
@@ -1,36 +1,50 @@
 """Tool for the identification of prompt injection attacks."""
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
-from langchain.pydantic_v1 import Field
+from langchain.pydantic_v1 import Field, root_validator
 from langchain.tools.base import BaseTool
 
 if TYPE_CHECKING:
     from transformers import Pipeline
 
 
-def _model_default_factory() -> Pipeline:
+def _model_default_factory(
+    model_name: str = "deepset/deberta-v3-base-injection"
+) -> Pipeline:
     try:
         from transformers import pipeline
     except ImportError as e:
         raise ImportError(
             "Cannot import transformers, please install with "
             "`pip install transformers`."
         ) from e
-    return pipeline("text-classification", model="deepset/deberta-v3-base-injection")
+    return pipeline("text-classification", model=model_name)
 
 
 class HuggingFaceInjectionIdentifier(BaseTool):
-    """Tool that uses deberta-v3-base-injection to detect prompt injection attacks."""
+    """Tool that uses HF model to detect prompt injection attacks."""
 
     name: str = "hugging_face_injection_identifier"
     description: str = (
         "A wrapper around HuggingFace Prompt Injection security model. "
         "Useful for when you need to ensure that prompt is free of injection attacks. "
         "Input should be any message from the user."
     )
-    model: Pipeline = Field(default_factory=_model_default_factory)
+    model: Any = Field(default_factory=_model_default_factory)
```

Comment:
would it be better to separate out the `model_name` and `model` params? Looks like it might be doing double duty currently.

Reply:
it'd be cleaner code but seems like a potentially confusing UX to me